R/comparePlot.R

```diff
@@ -226,72 +213,141 @@ comparePlot <- function(compareDF,
-                                yoffset = 0.05)
-    }
+        if (!is.null(referenceLine)) {
+            referenceLine <- paste(c("rgba(", paste(c(paste(col2rgb(referenceLine, alpha = FALSE), collapse = ","), 0.5), collapse = ","), ")"), collapse = "")
+            decorations <- list(line = append(decorations$line, list(list(color = referenceLine,
+                                                                          width = refLineThickness,
+                                                                          x     = -10,
+                                                                          x2    = 10,
+                                                                          y     = -10,
+                                                                          y2    = 10)
+            )))
+        }
 
+        if (missing(footnote)) {
+            footnote <- NULL
+        }
+
+        if (sigMeasurePlot) {
+            cx.data <- data.frame(a = round(compareDF[, 1], digits = 2), b = round(compareDF[, 2], digits = 2))
+            colnames(cx.data) <- c(xlabel, ylabel)
+            rownames(cx.data) <- rownames(compareDF)
+            var.annot <- data.frame(Group = compareDF$group)
+            colnames(var.annot) <- c("Group")
+            rownames(var.annot) <- rownames(cx.data)
+
+            CompPlot <-canvasXpress(data                    = cx.data,
+                                    varAnnot                = var.annot,
+                                    decorations             = decorations,
+                                    graphType               = "Scatter2D",
+                                    colorBy                 = "Group",
+                                    colors                  = symbolFill[c(1,4,2,3)],
+                                    legendPosition          = legendPosition,
+                                    scatterAxesEqual        = TRUE,
+                                    showDecorations         = TRUE,
+                                    sizeBy                  = "Group",
+                                    sizes                   = as.numeric(symbolSize[c(1,4,2,3)]),
+                                    sizeByShowLegend        = FALSE,
+                                    title                   = title,
+                                    subtitleScaleFontFactor = 0.5,
+                                    showAnimation           = FALSE,
+                                    width                   = "100%",
+                                    xAxis                   = list(xlab),
+                                    yAxis                   = list(ylab),
+                                    theme                   = theme,
+                                    citation                = footnote,
+                                    citationFontSize        = footnoteSize,
+                                    citationColor           = "black")
+        } else {
```

Comment:
switch these, so that the CXplot is first, and the ggplot is the 'else', cx should also be the default.